.github/workflows/labels.yml

```
@@ -2,18 +2,22 @@ name: Check labels
 
 on:
     pull_request:
+        types:
+            - opened
+            - labeled
+            - unlabeled
         branches:
             - main
 
 jobs:
     check_labels:
         runs-on: ubuntu-latest
         steps:
-          - name: 'Required "doc_required" or "doc_not_required" label'
+          - name: 'Required "doc required" or "doc not required" label'
             if: >
                 contains(github.event.pull_request.labels.*.name, 'doc required') == false &&
                 contains(github.event.pull_request.labels.*.name, 'doc not required') == false
-            run: exit 1
+            run: echo "::error Add \"doc required\" or \"doc not required\" label"
```

You must do both: Output the error message _and_ exit with a non-zero code
```suggestion
            run: |
              echo "::error Add \"doc required\" or \"doc not required\" label"
              exit 1
```